rxbinding/src/main/java/com/jakewharton/rxbinding2/internal/Preconditions.java

```diff
@@ -27,6 +30,15 @@ public static void checkArgument(boolean assertion, String message) {
     return value;
   }
 
+  public static boolean isNotOnMainThread(Observer<?> observer) {
```

Comment:
I would split it to two methods.
`isMainThread` which would only check
and second which would send error to observer.